File: FluentFTP.Examples/SetHashAlgorithm.cs
using System;
using System.Net;
using System.Threading;
using System.Threading.Tasks;
using FluentFTP;

namespace Examples {
	internal static class SetHashAlgorithmExample {

		public static void SetHashAlgorithm() {
			using (var conn = new FtpClient("127.0.0.1", "ftptest", "ftptest")) {
				conn.Connect();

				if (conn.HashAlgorithms.HasFlag(FtpHashAlgorithm.MD5)) {
					conn.SetHashAlgorithm(FtpHashAlgorithm.MD5);
				}
			}
		}

		public static async Task SetHashAlgorithmAsync() {
			var token = new CancellationToken();
			using (var conn = new FtpClient("127.0.0.1", "ftptest", "ftptest")) {
				await conn.ConnectAsync(token);

				if (conn.HashAlgorithms.HasFlag(FtpHashAlgorithm.MD5)) {
					await conn.SetHashAlgorithmAsync(FtpHashAlgorithm.MD5);
				}
			}
		}

	}
}

File: FluentFTP.VBExamples/SetHashAlgorithm.vb
Imports System
Imports System.Net
Imports System.Threading
Imports System.Threading.Tasks
Imports FluentFTP

Namespace Examples
	Friend Module SetHashAlgorithmExample
		Sub SetHashAlgorithm()
			Using conn = New FtpClient("127.0.0.1", "ftptest", "ftptest")
				conn.Connect()

				If conn.HashAlgorithms.HasFlag(FtpHashAlgorithm.MD5) Then
					conn.SetHashAlgorithm(FtpHashAlgorithm.MD5)
				End If
			End Using
		End Sub

		Async Function SetHashAlgorithmAsync() As Task
			Dim token = New CancellationToken()

			Using conn = New FtpClient("127.0.0.1", "ftptest", "ftptest")
				Await conn.ConnectAsync(token)

				If conn.HashAlgorithms.HasFlag(FtpHashAlgorithm.MD5) Then
					Await conn.SetHashAlgorithmAsync(FtpHashAlgorithm.MD5)
				End If
			End Using
		End Function
	End Module
End Namespace
